vm/vm.go

```diff
@@ -324,6 +323,14 @@ func (vm *VM) Initialize(
 
 	vm.mempool = mempool.New[*chain.Transaction](vm.tracer, vm.config.MempoolSize, vm.config.MempoolSponsorSize)
 
+	vm.builder = builder.NewTime(toEngine, snowCtx.Log, vm.mempool, func(t int64) (int64, int64, error) {
```

Comment:
Hmm moving this below where we apply options means that we will set the builder to this value even if the option to override the builder is specified. Moving this back above options causes the integration tests (which use the manual builder) to fail.

Seems this change may have introduced a bug in the manual builder that is covered up by this change. We could use `builder.NewTime` rather than manual implementation (clearly we don't need the manual implementation).